eth/eventdatahandler/event_data_handler.go

```diff
@@ -276,22 +250,124 @@ func (edh *EventDataHandler) processEvent(txn eventdb.RW, event ethtypes.Log) (T
 			return nil, fmt.Errorf("handle FeeRecipientAddressUpdated: %w", err)
 		}
 
-		task := func() error {
-			if !updated {
-				return nil
-			}
+		if !updated {
+			return &Task{}, fmt.Errorf("Provided receipeint address is the same")
+		}
+
+		task := NewTask(EventType(6), edh, event, nil)
+
+		return task, nil
 
-			if err := edh.taskExecutor.UpdateFeeRecipient(feeRecipientAddressUpdatedEvent); err != nil {
-				return fmt.Errorf("update fee recipient: %w", err)
+	default:
+		edh.logger.Warn("unknown event name", fields.Name(abiEvent.Name))
+		return nil, nil
+	}
+}
+
+// find and remove opposite tasks (start-stop, stop-start, liquidate-reactivate, reactivate-liquidate)
+// find superseding tasks and remove superseded ones (updateFee-updateFee)
+func cleanTaskList(tasks []*Task) []*Task {
+	taskMap := make(map[*Task]bool)
+	var resTask []*Task
+	for _, task := range tasks {
+		if _, exist := taskMap[task]; !exist {
+			taskMap[task] = true
+		}
+	}
+	for i, task := range tasks {
```

Comment:
Sorry, I didn't describe this in the TODO but we need to remove opposite/superseding tasks only if their "target" (validator/cluster) is same